src/main/java/seedu/address/storage/JsonTranscriptDeserializer.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.storage;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.ReadOnlyTranscript;
+import seedu.address.model.Transcript;
+import seedu.address.model.module.Code;
+import seedu.address.model.module.Credit;
+import seedu.address.model.module.Grade;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.Semester;
+import seedu.address.model.module.Year;
+
+/**
+ * Deserializer for {@link seedu.address.model.ReadOnlyTranscript}.
+ */
+public class JsonTranscriptDeserializer extends StdDeserializer<ReadOnlyTranscript> {
+
+
+    private static final Logger logger = LogsCenter.getLogger(JsonTranscriptStorage.class);
+
+    public JsonTranscriptDeserializer(Class<ReadOnlyTranscript> vc) {
+        super(vc);
+    }
+
+    public JsonTranscriptDeserializer() {
+        super(ReadOnlyTranscript.class);
+    }
+
+    @Override
+    public ReadOnlyTranscript deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+
+        Transcript transcript = new Transcript();
+        JsonNode node = jp.getCodec().readTree(jp);
+        try {
+            Iterator<JsonNode> elements = node.get("modules").get("internalList").elements();
+            while (elements.hasNext()) {
+                JsonNode e = elements.next();
```

Comment:
Should we give this a better name instead of `e`?

Reply:
Sure thing! Done.